### PR TITLE
DS-3307 -  Remove oss-parent and replace with release settings

### PR DESCRIPTION
**Pull request by tdonohue**
https://jira.duraspace.org/browse/DS-3307

* Removed Sonatype `oss-parent` and instead embedded release settings in this project's POM.  This syncs the POM up with that in `DSpace/DSpace` in this PR: https://github.com/DSpace/DSpace/pull/1367
* Also updated README and moved to README.md

Only light testing so far. I successfully ran a `dryRun` release.  A real release is also warranted once this code is merged.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -29,18 +33,275 @@
...
+                </plugins>
+            </build>
+        </profile>
+    </profiles>
+
     <!--
-      The SCM repository location is used by Continuum to update against
-      when changes have occurred.  This spawns a new build cycle and releases
-      snapshots into the snapshot repository below.
+     The SCM repository location is used by Continuum to update against
```
Comment (mwoodiupui) [inline pom.xml]: Uh, "Continuum"?  Officially retired, and though DSpace may have used it, it's been so long that I don't recall it.  Maybe just remove the coment altogether.

File: pom.xml [hunk 2]
```diff
@@ -29,18 +33,275 @@
...
+                        <groupId>org.sonatype.plugins</groupId>
+                        <artifactId>nexus-staging-maven-plugin</artifactId>
+                        <version>1.6.7</version>
+                        <extensions>true</extensions>
+                        <configuration>
+                            <!-- In your settings.xml, your username/password
+                                 MUST be specified for server 'ossrh' -->
+                            <serverId>ossrh</serverId>
+                            <nexusUrl>https://oss.sonatype.org/</nexusUrl>
+                            <autoReleaseAfterClose>true</autoReleaseAfterClose>
```
Comment (mwoodiupui) [inline pom.xml]: Should this be false?  Or will the change in behavior be noted in the release process?

Comment (pnbecker) [inline pom.xml]: Good catch!

File: pom.xml (hunk 1 above)
Comment (tdonohue) [inline pom.xml]: looks like this comment is also in DSpace/DSpace POM. This was a copy & paste thing

File: pom.xml (hunk 2 above)
Comment (tdonohue) [inline pom.xml]: This was copied from the DSpace/DSpace POM. So, if we change it here, we should change it both places: https://github.com/DSpace/DSpace/blob/master/pom.xml#L742

Comment (tdonohue) [inline pom.xml]: Fixed in latest commit https://github.com/DSpace/dspace-api-lang/pull/19/commits/aec97f7dc285860028a2ff3a5ccc0e17604f75c6

File: pom.xml (hunk 1 above)
Comment (tdonohue) [inline pom.xml]: Fixed in latest commit: https://github.com/DSpace/dspace-api-lang/pull/19/commits/aec97f7dc285860028a2ff3a5ccc0e17604f75c6

**Comment by tdonohue** [comment]
Merging to test release

**Pull request merged**